### PR TITLE
Added function remember to trim zeroes

### PR DESCRIPTION
**Pull request by DripYeager**
## Pull Request (PR) Checklist
Please check if your pull request fulfills the following requirements:
- [ ] The PR is submitted to the `main` branch.
- [ ] I've read the [Contribution Guidelines](https://github.com/Ivy-Apps/ivy-wallet/blob/main/CONTRIBUTING.md) and my PR doesn't break the rules.
- [ ] I've read the [Architecture Guidelines](https://github.com/Ivy-Apps/ivy-wallet/blob/main/docs/Architecture.md).
- [ ] I confirm that I've run the code locally and everything works as expected.
- [ ] 🎬 I've attached a **screen recoding** of the changes. 

> Tip: drag & drop the video to the PR description.

## What's changed?

Describe with a few bullets **what's new:**
- a
- b
- c

> 💡 Tip: Please, attach screenshots and screen recordings. It helps a lot!

## Risk Factors

**What may go wrong if we merge your PR?**

- a
- b

**In what cases your code won't work?**

- a
- b

## Does this PR closes any GitHub Issues?

Check **[Ivy Wallet Issues](https://github.com/Ivy-Apps/ivy-wallet/issues)**.

- Closes #ISSUE_NUMBER

> Replace `ISSUE_NUMBER` with the number of the GitHub issue that this PR is fixing. If you've done that correctly, you'll see the issue title linked when previewing your PR description.

> **!Note:** Do **not** link the PR number. Link the number/id of the GitHub Issue from [issues](https://github.com/Ivy-Apps/ivy-wallet/issues).

## Troubleshooting CI failures ❌

GitHub Actions failing? Read our [CI Troubleshooting guide](https://github.com/Ivy-Apps/ivy-wallet/blob/main/docs/CI-Troubleshooting.md).

### CODE REVIEW

File: widget/balance/src/main/java/com/ivy/widget/balance/WalletBalanceWidget.kt [hunk 1]
```diff
@@ -28,10 +29,19 @@ import kotlinx.coroutines.MainScope
 import kotlinx.coroutines.launch
 import javax.inject.Inject
 
+const val THOUSAND = 1000
+
 class WalletBalanceWidget(
     private val getAppStarter: () -> AppStarter,
 ) : GlanceAppWidget() {
-
+    private fun remember(balance: String): String {
```
Comment (ILIYANGERMANOV) [inline widget/balance/src/main/java/com/ivy/widget/balance/WalletBalanceWidget.kt:37]: Hey, you need to use Compose `remember` your remember doesn't memoize

File: widget/balance/src/main/java/com/ivy/widget/balance/WalletBalanceWidget.kt [hunk 2]
```diff
@@ -28,10 +29,19 @@ import kotlinx.coroutines.MainScope
...
+        val balanceDouble = balance.toDouble()
+        return if (Math.abs(balanceDouble) < THOUSAND) {
+            DecimalFormat("###,###.##").format(balanceDouble)
+        } else {
+            com.ivy.legacy.utils.shortenAmount(balanceDouble)
```
Comment (ILIYANGERMANOV) [inline widget/balance/src/main/java/com/ivy/widget/balance/WalletBalanceWidget.kt:42]: Import it properly

File: widget/balance/src/main/java/com/ivy/widget/balance/WalletBalanceWidget.kt (hunk 1 above)
Comment (ILIYANGERMANOV) [reply widget/balance/src/main/java/com/ivy/widget/balance/WalletBalanceWidget.kt:37]: @DripYeager to give you a hint:
```kotlin
val balanceFormatted = remember(balance) {
   // all of your code here...
}
```

Comment (DripYeager) [reply widget/balance/src/main/java/com/ivy/widget/balance/WalletBalanceWidget.kt:37]: thanks for the help, i have updated the code let me if still there is any issue

File: widget/balance/src/main/java/com/ivy/widget/balance/WalletBalanceWidget.kt [hunk 3]
```diff
@@ -28,10 +32,23 @@ import kotlinx.coroutines.MainScope
...
 import javax.inject.Inject
 
+const val THOUSAND = 1000
+
 class WalletBalanceWidget(
     private val getAppStarter: () -> AppStarter,
 ) : GlanceAppWidget() {
-
+    @Composable
+    fun trimzero(balance: String): String {
```
Comment (ILIYANGERMANOV) [inline widget/balance/src/main/java/com/ivy/widget/balance/WalletBalanceWidget.kt:41]: rename this function to `formatBalance`

**Review by ILIYANGERMANOV — APPROVED** [review]
Fix my last comment and the CI and then we can merge ✅ Also make "Update branch"

File: widget/balance/src/main/java/com/ivy/widget/balance/WalletBalanceWidget.kt (hunk 3 above)
Comment (DripYeager) [reply widget/balance/src/main/java/com/ivy/widget/balance/WalletBalanceWidget.kt]: Done. you can go through it again

**Comment by ILIYANGERMANOV** [comment]
@DripYeager click Update Branch and we can merge

**Comment by DripYeager** [comment]
Thank you for being patient with me, it was my first time contributing. You were a great help.

**Pull request merged**